datafusion/src/physical_plan/mod.rs

```diff
@@ -152,6 +162,133 @@ pub trait ExecutionPlan: Debug + Send + Sync {
     fn metrics(&self) -> HashMap<String, SQLMetric> {
         HashMap::new()
     }
+
+    /// Format this `ExecutionPlan` to `f` in the specified type.
+    ///
+    /// Should not include a newline
+    ///
+    /// Note this function prints a placeholder by default to preserve
+    /// backwards compatibility.
+    fn fmt_as(&self, _t: DisplayFormatType, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "ExecutionPlan(PlaceHolder)")
+    }
+}
+
+/// Return a [wrapper](DisplayableExecutionPlan) around an
```

Comment:
This is the main proposed interface: `displayable` which returns a `struct` which then has several ways to display it. It would be ideal if I could add this to `ExecutionPlan` directly itself, but since it is a trait this was the best I could come up with (along with a bunch of documentation)